gtwrap/pybind_wrapper.py

```diff
@@ -254,37 +254,100 @@ def wrap_operators(self, operators, cpp_class, prefix='\n' + ' ' * 8):
+            cpp_class = class_name + "::" + cpp_class
+
+        res = '{prefix}py::enum_<{cpp_class}>({module}, "{enum.name}", py::arithmetic())'.format(
+            prefix=prefix, module=module, enum=enum, cpp_class=cpp_class)
+        for enumerator in enum.enumerators:
+            res += '\n{prefix}    .value("{enumerator.name}", {cpp_class}::{enumerator.name})'.format(
+                prefix=prefix, enumerator=enumerator, cpp_class=cpp_class)
+        res += ";\n\n"
+        return res
+
+    def wrap_enums(self, enums, instantiated_class, prefix=' ' * 4):
+        """Wrap multiple enums defined in a class."""
+        cpp_class = instantiated_class.cpp_class()
+        module_var = instantiated_class.name.lower()
+        res = ''
+
+        for enum in enums:
+            res += "\n" + self.wrap_enum(
+                enum,
+                class_name=cpp_class,
+                module=module_var,
+                prefix=prefix)
+        return res
+
     def wrap_instantiated_class(
             self, instantiated_class: instantiator.InstantiatedClass):
         """Wrap the class."""
         module_var = self._gen_module_var(instantiated_class.namespaces())
         cpp_class = instantiated_class.cpp_class()
         if cpp_class in self.ignore_classes:
             return ""
-        return (
-            '\n    py::class_<{cpp_class}, {class_parent}'
-            '{shared_ptr_type}::shared_ptr<{cpp_class}>>({module_var}, "{class_name}")'
-            '{wrapped_ctors}'
-            '{wrapped_methods}'
-            '{wrapped_static_methods}'
-            '{wrapped_properties}'
-            '{wrapped_operators};\n'.format(
-                shared_ptr_type=('boost' if self.use_boost else 'std'),
-                cpp_class=cpp_class,
-                class_name=instantiated_class.name,
-                class_parent="{instantiated_class.parent_class}, ".format(
-                    instantiated_class=instantiated_class)
-                if instantiated_class.parent_class else '',
-                module_var=module_var,
-                wrapped_ctors=self.wrap_ctors(instantiated_class),
-                wrapped_methods=self.wrap_methods(instantiated_class.methods,
-                                                  cpp_class),
-                wrapped_static_methods=self.wrap_methods(
-                    instantiated_class.static_methods, cpp_class),
-                wrapped_properties=self.wrap_properties(
-                    instantiated_class.properties, cpp_class),
-                wrapped_operators=self.wrap_operators(
-                    instantiated_class.operators, cpp_class)))
+        if instantiated_class.parent_class:
+            class_parent = "{instantiated_class.parent_class}, ".format(
+                instantiated_class=instantiated_class)
+        else:
+            class_parent = ''
+
+        if instantiated_class.enums:
+            # If class has enums, define an instance and set module_var to the instance
```

Comment:
why not just name all the classes?  Is there any downside to giving the class a variable name?

Reply:
There isn't other than a lack of time. I want to push this out fast since there are other things that need my attention...